x-pack/plugins/endpoint/server/routes/resolver/utils/fetch.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IScopedClusterClient } from 'kibana/server';
+import { extractEntityID, extractParentEntityID } from './normalize';
+import { getPaginationParams } from './pagination';
+import { Tree } from './tree';
+import { LifecycleQuery } from '../queries/lifecycle';
+import { ChildrenQuery } from '../queries/children';
+import { EventsQuery } from '../queries/events';
+import { StatsQuery } from '../queries/stats';
+
+export class Fetcher {
+  constructor(
+    private readonly client: IScopedClusterClient,
+    private readonly id: string,
+    private readonly endpointID?: string
+  ) {}
+
+  public async ancestors(limit: number): Promise<Tree> {
+    const tree = new Tree(this.id);
+    await this.doAncestors(tree, this.id, this.id, limit);
+    return tree;
+  }
+
+  public async children(limit: number, generations: number, after?: string): Promise<Tree> {
+    const tree = new Tree(this.id);
+    await this.doChildren(tree, [this.id], limit, generations, after);
+    return tree;
+  }
+
+  public async events(limit: number, after?: string): Promise<Tree> {
+    const tree = new Tree(this.id);
+    await this.doEvents(tree, limit, after);
+    return tree;
+  }
+
+  public async alerts(limit: number, after?: string): Promise<Tree> {
```

Comment:
Not implemented yet, but same as the events.

Reply:
Same comment as with alerts--the alerts themselves are only relevant when you're drilling into the details of what a process actually did. We calculate stat summaries for every node, but we shouldn't be displaying alerts right from the get-go for all nodes--only when one is in focus.